packages/api-cli/src/api.ts

```diff
@@ -187,15 +201,30 @@ async function makeTx ({ fn, log }: CallInfo): Promise<void> {
   assert(CRYPTO.includes(sign), `The crypto type can only be one of ${CRYPTO.join(', ')} found '${sign}'`);
 
   const keyring = new Keyring();
-  const account = keyring.createFromUri(seed, {}, sign as 'ed25519');
+  let auth = keyring.createFromUri(seed, {}, sign as 'ed25519');
+
+  let signable;
+  if (sudo) {
+    const provider = new WsProvider(ws);
```

Comment:
Instead of doing this, I would suggest -

- just add the constructed api object on to `CallInfo`
- this way it is available already and we don't need to re-construct (it is a heavy operation since it will initialize and read metadata)